q_types.py

```diff
@@ -0,0 +1,216 @@
+            raise ValueError("Only 1 answer should be provided")
+        if question['answer'][0] not in ('true','false'):
+            raise ValueError("valid answers are true or false")
+
+    def bb(self):
+        return [self.type, self.prompt, self.question['answer'][0]]
+
+    def latex(self):
+        return [('answer', self.question['answer'][0])]
+
+class ESS(Q_process):
+    def bb(self):
+        if 'example' in self.question.keys():
+            return [self.type, self.prompt, self.question['example'][0]]
+        else:
+            return [self.type, self.prompt]
+
+    def latex(self):
+        if 'example' in self.question.keys():
+            return [('example', self.question['example'][0])]
+        else:
+            return None
+
+class ORD(Q_process):
+    def __init__(self, question):
+        Q_process.__init__(self, question)
+        if not 1 <= len(question['answer']) <= 20:
+            raise AssertionError("number of answers must be between 1 and 20")
+
+    def bb(self):
+        return [self.type, self.prompt, *self.question['answer']]
+
+    def latex(self):
+        items = [(num, ans) for num, ans in enumerate(self.question['answer'],1)]
+        return items
+
+class MAT(Q_process):
+    def __init__(self, question):
+        Q_process.__init__(self, question)
+        if len(question['match_a']) != len(question['match_b']):
+            raise AssertionError("All options must have matching answers")
+        a, b = question['match_a'], question['match_b']
+        self.merged_pairs = [sub[item] for item in range(len(b)) for sub in [a, b]]
+
+    def bb(self):
+        return [self.type, self.prompt, *self.merged_pairs]
+
+    def latex(self):
+        letter = ('a','b')
+        items = [(str(num//2)+letter[num%2], ans) for num, ans in
+                zip(range(2,len(self.merged_pairs)+2), self.merged_pairs)]
+        return items
+
+class FIL(Q_process):
+    def bb(self):
+        return [self.type, self.prompt]
+
+    def latex(self):
+        return None
+
+class NUM(Q_process):
+    def __init__(self, question):
+        Q_process.__init__(self, question)
+        try:
+            float(question['answer'][0])
+            self.ans = question['answer'][0]
+            if 'tolerance' in question.keys():
+                float(question['tolerance'][0])
+                self.tol = question['tolerance'][0]
+        except:
+            raise ValueError('answer and tolerance must be numbers')
+        
+    def bb(self):
+        if 'tolerance' in self.question.keys():
+            return [self.type, self.prompt, self.ans, self.tol]
+        else:
+            return [self.type, self.prompt, self.ans]
+
+    def latex(self):
+        if 'tolerance' in self.question.keys():
+            return [('answer', self.ans), ('tolerance', r'$\pm$'+self.tol)]
+        else:
+            return [('answer', self.ans)]
+class SR(ESS):
+    """Inherets functions form essay question class as they are identical"""
```

Comment:
There should be 2 blank lines above this class. inherets -> inherits

The docstring is more of a code comment than documentation. Better just to use a comment